synapse/membrane.py

```diff
@@ -0,0 +1,156 @@
+# FIXME support ival splices
+# FIXME implement roles/groups/users/perms
+
+import copy
+import fnmatch
+import logging
+import operator
+
+from synapse.eventbus import EventBus
+import synapse.lib.syntax as s_syntax
+
+logger = logging.getLogger(__name__)
+
+class Membrane(EventBus):
+
+    _SPLICE_NAMES = (
+        'node:add',
+        'node:del',
+        'node:prop:set',
+        'node:prop:del',
+        'node:tag:add',
+        'node:tag:del',
+        #'node:ival:set',
+        #'node:ival:del',
+    )
+    _DEFAULT_RULE = [{'query': '', 'insts': []}]
```

Comment:
Talked to @invisig0th about this. Probably going to change back and not use s_syntax.